case-lib/lib.sh

```diff
@@ -1352,3 +1352,27 @@ reboot_wrapper()
     echo "Reboot command" && sleep 5
     sudo reboot now
 }
+
+# Play sound and record it
+# Parameters: 1-arecord options 2-aplay options
+play_and_record()
+{
+    dlogi "Play [aplay $2] and capture sound [arecord $1]"
+    arecord $1 & PID=$!
+    sleep 1
+    aplay $2
```

Comment:
Since bash version 4.3 (2014), you can pass the (name of) arrays to functions thanks to "nameref"

https://stackoverflow.com/questions/16461656/how-to-pass-array-as-an-argument-to-a-function-in-bash 
https://mywiki.wooledge.org/BashGuide/Arrays
https://www.gnu.org/software/bash/manual/html_node/Shell-Parameters.html

```bash
sub()
{
    local -n rec_opts_nameref="$1" # -n is nameref
    local -n play_opts_nameref="$2"

    arecord "${rec_opts_nameref[@]}"

}

array_with_special_chars=("A B" "C*")
arr2=("A B" "C*")

sub array_with_special_chars arr2

```